REST-SDK/test/Cumulocity.SDK.Client.UnitTest/Messaging.Notifications/NotificationSubscriptionApiImplTest.cs

```diff
@@ -129,5 +174,74 @@ public void testSubscribeWithInvalidHost()
                 Assert.Throws<UriFormatException>(() => restConnector.Object.Post<NotificationSubscriptionRepresentation>(It.IsAny<string>(), It.IsAny<CumulocityMediaType>(), It.IsAny<NotificationSubscriptionRepresentation>()));
             }
         }
+
+        [Fact]
+        public void testGetSubscriptionsWithoutFilter()
+        {
+            Assert.NotNull(api.getSubscriptionsByFilter(null));
+        }
+
+        [Fact]
+        public void testGetSubscriptionsWithFilter()
+        {
+            NotificationSubscriptionFilter filter = new NotificationSubscriptionFilter().byContext(SubscriptionContext.TENANT.ToDescriptionString());
+            Assert.Equal("tenant", filter.getContext());
+            Assert.Null(filter.getSource());
+            Assert.NotNull(api.getSubscriptionsByFilter(filter));
+        }
+
+        [Fact]
+        public void testNotificationSubscriptionCollectionRepresentationJson()
+        {
+            NotificationSubscriptionRepresentation subscription = new NotificationSubscriptionRepresentation();
+            NotificationSubscriptionRepresentation subscription2 = new NotificationSubscriptionRepresentation();
+            subscription2.Context = "sampleContext";
+            subscription2.Subscription = "tenantSub";
+            subscription2.FragmentsToCopy = new List<string>() { "c8y_Temparature", "c8y_humidity" };
+            NotificationSubscriptionCollectionRepresentation notificationSubscriptionCollectionRepresentations = new NotificationSubscriptionCollectionRepresentation();
+            notificationSubscriptionCollectionRepresentations.Subscriptions = new List<NotificationSubscriptionRepresentation>() { subscription, subscription2 };
+            string jsonRepresentation = @"[{},{""Context"":""sampleContext"",""Subscription"":""tenantSub"",""FragmentsToCopy"":[""c8y_Temparature"",""c8y_humidity""]}]";
+            Assert.Equal(jsonRepresentation, JsonConvert.SerializeObject(notificationSubscriptionCollectionRepresentations));
+            List<NotificationSubscriptionRepresentation> deserializedRepresentation = JsonConvert.DeserializeObject<List<NotificationSubscriptionRepresentation>>(jsonRepresentation);
+            Assert.Equal(notificationSubscriptionCollectionRepresentations.Subscriptions.Count, deserializedRepresentation.Count);
+            Assert.Equal(notificationSubscriptionCollectionRepresentations.Subscriptions[1].Context, deserializedRepresentation[1].Context);
+            Assert.Equal(notificationSubscriptionCollectionRepresentations.Subscriptions[1].Subscription, deserializedRepresentation[1].Subscription);
+        }
+
+        [Fact]
+        public void testNotificationSubscriptionFilterRepresentationJson()
+        {
+            string filterRepresentationJson = @"{""apis"":[""inventoryApi"", ""alarmApi""], ""typeFilter"":""mo""}";
```

Comment:
Json serialization/deserialization tests for Representation classes which have jsonproperty